extension/chrome/elements/compose-modules/compose-storage-module.ts

```diff
@@ -81,19 +81,47 @@ export class ComposeStorageModule extends ViewModule<ComposeView> {
     return await PassphraseStore.get(this.view.acctEmail, senderKi.fingerprints[0]);
   }
 
-  public lookupPubkeyFromDbOrKeyserverAndUpdateDbIfneeded = async (email: string, name: string | undefined): Promise<Contact | "fail"> => {
+  public lookupPubkeyFromKeyserversThenOptionallyFetchExpiredByFingerprintAndUpsertDb = async (
+    email: string, name: string | undefined
+  ): Promise<Contact | "fail"> => {
+    // note by Tom 2021-08-10: This line below looks weird in the context of recently allowed
+    //    more than one public key per recipient. We are blindly getting the first public key,
+    //    and making decisions based on that. We should be using the whole array instead.
```

Comment:
This is not "blindly" the first public key. The compatibility `get` method is designed to provide the "best" key, that is non-expired (if possible) and valid as long as possible. Of course, we can replace `get` calls with `getOneWithAllPubkeys` if required.

Reply:
Got it, the key is being selected carefully to be useful. 

But it's still just one key, when we should be dealing with arrays. This has side effects. For example, when I hover over a recipient, it says `Does use encryption` and then shows only one key, but it should really be showing all public key that we will encrypt for.

I'm thinking we could use `getOneWithAllPubkeys`, which would indeed return all pubkeys including expired and revoked ones. And then have the key selection logic elsewhere.

It seems this public key selection logic is only used in compose view anyway, so we could move it there?

This means we could completely deprecate the `get` method that gets by email. I don't think there is any valid situation where we want to only get one public key anymore, except for legacy code that should be updated to work with an array meaningfully.

Reply:
Sure, we can create an issue to work in this direction.

Reply:
:+1: I'm filing https://github.com/FlowCrypt/flowcrypt-browser/issues/3905 , marking this as resolved